test/object-store/dictionary.cpp

```diff
@@ -937,7 +937,71 @@ TEMPLATE_TEST_CASE("dictionary of objects", "[dictionary][links]", cf::MixedVal,
         Obj target_obj = target->create_object().set(col_target_value, T(values[i]));
         dict.insert(keys[i], target_obj);
     }
+    r->commit_transaction();
 
+    SECTION("callback with empty keypatharray") {
```

Comment:
Running these tests on every property type doesn't actually hit any code paths that aren't hit by just running them on one, so that's just adding test run time for little benefit. It should instead be in a separate test case.